docs/se/openapi-geneator.adoc

```diff
@@ -0,0 +1,218 @@
+    <artifactId>openapi-generator-maven-plugin</artifactId>
+    <version>${openapi-generator-version}</version>
+</dependency>
+----
+
+== Configuration
+
+To customize generated project different options can be used.
+Options for OpenAPI Generator are the same whether you're using the Maven/Gradle Plugins, CLI, or Online generation.
+
+General Configuration parameters
+
+At a minimum generation of the new Helidon MP project requires:
+
+[cols="1,1"]
+|===
+|Option | Description
+
+|inputSpec
+|OpenAPI Spec file path
+|generatorName
+|target generator name
+|library
+|library template (sub-template)
+|===
+
+The whole list of the general configuration parameters for OpenAPI Generator is available on link:https://github.com/OpenAPITools/openapi-generator/tree/master/modules/openapi-generator-maven-plugin[openapi-generator-maven-plugin] page.
+
+Predefined values for required options for Helidon SE Generators :
+
+[cols="1,1,1"]
+|===
+|Option |Value |Description
+
+|generatorName
+|java-helidon-server
+|generator name for Helidon SE Server
+
+|generatorName
+|java-helidon-client
+|generator name for Helidon SE Client
+
+|library
+|se
+|used library for Helidon SE Server/Client
+|===
+
+OpenAPI Generators for Helidon SE support such specific options :
+
+[cols="1,1,1,1,1"]
+|===
+|Option |Description |Values |Default |Notes
+
+|helidonVersion
+|Version of Helidon
+|
+|2.5.2
+|
+
+|fullProject
+|Generate API files only
+|true/false
+|false
+| API files : API Interfaces or abstract classes, models (files we do not expect the end user to modify)
+
+|serializationLibrary
+|Library to use for serializing JSON
+|jsonb, jackson
+|jackson
+|
+|===
+
+== Usage
+
+CLI, Maven/Gradle Plugins, or Online generation can be used to generate Helidon SE project.
+
+To generate a project using CLI at first `openapi-generator-cli.jar` or `Launcher Script` should be downloaded (details can be found in link:https://github.com/OpenAPITools/openapi-generator#1---installation[OpenAPI Generator GitHub Repository])
+
+To generate Helidon SE server stubs for petstore.yaml (OpenApi Document) using `openapi-generator-cli.jar`, please run the following :
+
+[source,bash]
+----
+java -jar openapi-generator-cli.jar \
+  generate \
+  -i petstore.yaml \
+  -g java-helidon-server \
+  --library se \
+  -p groupId=test.se.helidon \
+  -p artifactId=test-helidon-se-server \
+  -p artifactVersion=1.0.0-SNAPSHOT
+----
+
+This command will generate maven project for Helidon SE Server with groupId, artifactId and version as described above.
+
+To generate SE REST client for petstore.yaml (OpenApi Document) using `openapi-generator-cli.jar`, please run the following :
+
+[source,bash]
+----
+java -jar openapi-generator-cli.jar \
+  generate \
+  -i petstore.yaml \
+  -g java-helidon-client \
+  --library se
+----
+
+This command will generate maven project for Helidon SE Rest Client that can interact with server which exposes API that described in the OpenAPI document (petstore.yaml).
```

Comment:
This command will generate Maven project for a Helidon SE Rest client that can interact with the server that exposes the API as described in the OpenAPI document petstore.yaml.